### PR TITLE
Implement Webpack plugin

### LINKED ISSUE #392: Implement WebPack Snaps plugin

### PR DESCRIPTION
**Pull request by Mrtenz**
Closes #392. Related to #389.

Similar to #410, this PR adds a new package which is a Webpack plugin to apply the code processing to fix issues with SES.

### CODE REVIEW

File: packages/plugin-webpack/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# @metamask/snap-webpack
```
Comment (rekmarks) [inline packages/plugin-webpack/README.md:1]: Let's rename it per #421:
```suggestion
# @metamask/snaps-webpack-plugin
```

File: packages/plugin-webpack/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { postProcess, PostProcessOptions } from '@metamask/snap-utils';
+import { Compiler, WebpackPluginInstance } from 'webpack';
+import { RawSource } from 'webpack-sources';
+
+const PLUGIN_NAME = 'SnapWebpackPlugin';
```
Comment (rekmarks) [inline packages/plugin-webpack/src/plugin.ts:5]: To add further context for this: the branding we've landed on is "MetaMask Snaps", so we should use "Snaps" in the name for packages / things intended for external audiences:

```suggestion
const PLUGIN_NAME = 'SnapsWebpackPlugin';
```

File: packages/plugin-webpack/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  public readonly options: Partial<Options>;
+
+  constructor(options: Partial<Options> = {}) {
+    this.options = options;
+  }
+
+  apply(compiler: Compiler) {
+    compiler.hooks.compilation.tap(PLUGIN_NAME, (compilation) => {
+      compilation.hooks.processAssets.tap(PLUGIN_NAME, (assets) => {
+        Object.keys(assets).forEach((assetName) => {
```
Comment (rekmarks) [inline packages/plugin-webpack/src/plugin.ts:19]: Are the assets guaranteed to be JavaScript files here?

Comment (Mrtenz) [reply packages/plugin-webpack/src/plugin.ts:19]: Good question! I'll add a test case which processes other files, and update this code if necessary.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**